integration-tests/src/test/scala/za/co/absa/spline/BasicIntegrationTests.scala

```diff
@@ -117,15 +122,41 @@ class BasicIntegrationTests extends AsyncFlatSpec
               .write.mode(Append).saveAsTable(tableName)
           }
 
-          (plan2, _) <- captor.lineageOf {
+          // Spark 3.4+ is creating 2 commands for both writes here so we need to ignore one
+          // We only want the one that is from CreateDataSourceTableAsSelectCommand
+          // The one we ignore here is an extra InsertIntoHadoopFsRelationCommand
+          // They can come out of order so we need to filter out which one is which.
+          (plan2, _) <- if (ver"$SPARK_VERSION" >= ver"3.4.0") {
+            captor.lineageOf {
+              Thread.sleep(5000)
+            }
+          } else Future[(ExecutionPlan, Seq[ExecutionEvent])](null, null)
+
+          (plan3, _) <- captor.lineageOf {
             spark
               .read.table(tableName)
               .write.mode(Overwrite).saveAsTable("somewhere")
           }
+
+          (plan4, _) <- if (ver"$SPARK_VERSION" >= ver"3.4.0") {
+            captor.lineageOf {
+              Thread.sleep(5000)
+            }
+          } else Future[(ExecutionPlan, Seq[ExecutionEvent])](null, null)
         } yield {
           println("yield")
-          val writeUri = plan1.operations.write.outputSource
-          val readUri = plan2.operations.reads.head.inputSources.head
+
+          val writePlan = Seq(plan1, plan2)
```

Comment:
There is some common code here on filtering out the correct options we want here between this and the Hive CTAS code - not sure how much level of DRY you all are looking to reduce here but this is potential.